kasa/iot/iotdevice.py

```diff
@@ -312,11 +312,13 @@ async def update(self, update_children: bool = True):
 
         await self._modular_update(req)
 
+        self._set_sys_info(self._last_update["system"]["get_sysinfo"])
+        for module in self._modules.values():
+            module._handle_update()
```

Comment:
This changes the common lib behavior, so it'd be better done in its own PR.

Reply:
https://github.com/python-kasa/python-kasa/pull/921